src/components/navbar.js

```diff
@@ -20,27 +46,45 @@ const Navbar = () => {
       setError(e.message)
     }
   }, [firebaseApp])
-
   return (
-    <Box
+    <Flex
       sx={{
         backgroundColor: 'background',
-        mb: 4,
+        alignContent: 'flex-start',
+        alignItems: 'center',
+        flexWrap: 'wrap',
+        justifyContent: 'flex-end',
+        fontSize: '24px',
+        maxWidth: '960px',
+        margin: 'auto',
+        padding: '5px'
       }}
     >
-      this will be replaced with the real navbar
-      {user === null ? (
-        <Link to='/login'>login and register</Link>
-      ) : (
-        <Box>
-          <Text>welcome {user.displayName}</Text>
-          <Link to='/' onClick={handleLogout}>
-            Logout
-          </Link>
-          <Text sx={{ color: 'highlight' }}>{error}</Text>
-        </Box>
-      )}
-    </Box>
+      <Box
+        sx={{
+          flexBasis: '7em',
+          marginRight: 'auto'
+        }}
+      >
+        <Image
+          src={logo}
+          alt='climatedu logo'
+        />
+      </Box>
+      {links.map(({location,text,icon,dropdown}) => {
+        const loc = location(user);
+        const active = window.location.pathname == loc;
```

Comment:
This should use the router's location matching instead of doing matching yourself.